ec2/floatingip.py

```diff
@@ -0,0 +1,95 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Boto Imports
+from boto.exception import EC2ResponseError
+from boto.exception import BotoServerError
+
+# Cloudify imports
+from cloudify import ctx
+from cloudify.exceptions import NonRecoverableError
+from cloudify.decorators import operation
+from ec2 import connection
+
+
+@operation
+def create(**kwargs):
+    """ Gets a floating IP from an Amazon Elastic IP.
+    """
+    ec2_client = connection.EC2ConnectionClient().client()
+    ctx.logger.debug('Getting Elastic IP from Amazon EC2.')
+
+    try:
+        address_object = ec2_client.allocate_address()
+    except (EC2ResponseError, BotoServerError) as e:
+        raise NonRecoverableError('(Node: {0}): Error. Failed to get '
+                                  'floating ip, returned: {1}.'
```

Comment:
Um.. this is kinda relevant for the rest of the review.. .but didn't we decide that all resources are named according to their Cloud environment?

We shouldn't have "floating ip" anywhere.. only Elastic IP. or .. am I wrong?